### ISSUE TITLE
RC2014 with SM Baker SIO-2 and CTC cards

### ISSUE THREAD

**Issue by MMaciocia** [wwarthen/RomWBW#194]
It may be that I am being stupid here, but, with CTC enabled, port address of 0x40 set, SIO-2 enabled with the SIOMODE_SMB set as TRUE, CTCMODE is CTCMODE_RC, IM-2 correctly selected, then at no point is the SIO-B channel baud rate set in the CTC. Have checked the TO1 output with a scope and even poked through the monitor correct values into the CTC channel B and scoped the output pre (DC level) and post, whatever clock I had programmed as a value. Also, tried using mode com1: 115200 etc and no change to the CTC channel B TO1. Triggers are correctly set to the system clock.
Am I missing something in the conditional assembly files? I have been editing RC2014_std and RC2014_kio .asm files for the kio edit as below, KIO disabled as none in my system:

#DEFINE		PLATFORM_NAME	"RC2014 (KIO)"
;
#include "Config/RCZ80_std.asm"
;
INTMODE		.SET	2			; INTERRUPTS: 0=NONE, 1=MODE 1, 2=MODE 2
;
KIOENABLE	.SET	FALSE		; ENABLE ZILOG KIO SUPPORT
;
CTCENABLE	.SET	TRUE		; ENABLE ZILOG CTC SUPPORT
CTCBASE		.SET	$40			; CTC BASE I/O ADDRESS
;
ACIAENABLE	.SET	FALSE		; ACIA: ENABLE MOTOROLA 6850 ACIA DRIVER (ACIA.ASM)
;
SIOENABLE	.SET	TRUE		; SIO: ENABLE ZILOG SIO SERIAL DRIVER (SIO.ASM)
SIOCNT		.SET	1			; SIO: NUMBER OF CHIPS TO DETECT (1-2), 2 CHANNELS PER CHIP
SIO0MODE	.SET	SIOMODE_SMB	; SIO 0: CHIP TYPE: SIOMODE_[RC|SMB|ZP|EZZ80]
SIO0BASE	.SET	$80			; SIO 0: REGISTERS BASE ADR
SIO0CTCC	.SET	0			; SIO 0: CTC CHANNEL CLOCK SCALER, (0-3), -1 FOR NONE
SIO0ACLK	.SET	7372800		; SIO 0A: OSC FREQ IN HZ, ZP=2457600/4915200, RC/SMB=7372800
SIO0BCLK	.SET	7372800		; SIO 0B: OSC FREQ IN HZ, ZP=2457600/4915200, RC/SMB=7372800

and for the RC2014_std settings as below (just left the ACIA setting TRUE here as it is set FALSE in kio file).

#include "cfg_rcz80.asm"
;
CPUOSC		.SET	7372800		; CPU OSC FREQ IN MHZ
;
ACIAENABLE	.SET	TRUE		; ACIA: ENABLE MOTOROLA 6850 ACIA DRIVER (ACIA.ASM)
;
SIOENABLE	.SET	TRUE		; SIO: ENABLE ZILOG SIO SERIAL DRIVER (SIO.ASM)
;
FDENABLE	.SET	FALSE		; FD: ENABLE FLOPPY DISK DRIVER (FD.ASM)
FDMODE		.SET	FDMODE_RCWDC	; FD: DRIVER MODE: FDMODE_[DIO|ZETA|DIDE|N8|DIO3]
;
IDEENABLE	.SET	TRUE		; IDE: ENABLE IDE DISK DRIVER (IDE.ASM)
;
PPIDEENABLE	.SET	TRUE		; PPIDE: ENABLE PARALLEL PORT IDE DISK DRIVER (PPIDE.ASM)

**Comment by wwarthen** [comment]
@MMaciocia, Looks like you closed the issue.  Were you able to resolve the problem?  Is there anything that needs to be addressed in RomWBW?

Thanks,

Wayne

**Comment by MMaciocia** [comment]
Yes, it was stupidity on my part that a trawl through the list output from the assembly found for me when I saw the actual hex port number that was assembled versus what I thought it was when I set it in the conditionals file! I had spent about 3 days mucking about with it, being able to directly set the CTC from the monitor and had not thought to check that the actual code that I used in monitor was happening in the ROM – just looking at something for too long to see the obvious!

 

Sorry for setting bells ringing!

 

Many thanks for the fabulous work you have done.

 

Best regards

Michael
